### PR TITLE
upgrades sdk and refactors to new rpc patterns

### PR DESCRIPTION
**Pull request by aristidesstaffieri**
Upgardes to p11 compatible soroban-client, refactors to new rpc patterns.

### CODE REVIEW

File: src/helpers/soroban.ts [hunk 1]
```diff
@@ -114,22 +106,12 @@ export const simulateTx = async <ArgType>(
...
-  let convertedScVal: any;
-  try {
-    // handle a case where scValToNative doesn't properly handle scvString
-    convertedScVal = scVal.str().toString();
-    return convertedScVal;
-  } catch (e) {
-    console.error(e);
+  const response = await server.simulateTransaction(tx);
+
+  if ("result" in response && response.result !== undefined) {
```
Comment (Shaptic) [inline src/helpers/soroban.ts:111]: you should use `SorobanRpc.isSimulationSuccess` here (there are also `Restore` and `Error` variants)

File: src/helpers/soroban.ts [hunk 2]
```diff
@@ -303,11 +286,11 @@ export const getEstimatedFee = async (
   const raw = tx.build();
 
   const simResponse = await server.simulateTransaction(raw);
-  if (simResponse.error) {
+  if ("error" in simResponse) {
     throw simResponse.error;
   }
 
-  if (!simResponse.results || simResponse.results.length < 1) {
+  if (!("result" in simResponse)) {
```
Comment (Shaptic) [inline src/helpers/soroban.ts:293]: same here re: using lib to distinguish

File: src/helpers/soroban.ts (hunk 1 above)
Comment (aristidesstaffieri) [reply src/helpers/soroban.ts:111]: ah nice, thanks

Comment (aristidesstaffieri) [reply src/helpers/soroban.ts]: fixed in 0ccba4ca0455f3704b295ad9ca5d0261afbc4f67

File: src/helpers/soroban.ts (hunk 2 above)
Comment (aristidesstaffieri) [reply src/helpers/soroban.ts]: fixed in 0ccba4ca0455f3704b295ad9ca5d0261afbc4f67

File: src/helpers/soroban.ts [hunk 3]
```diff
@@ -114,22 +106,15 @@ export const simulateTx = async <ArgType>(
...
+  const response = await server.simulateTransaction(tx);
+
+  if (
+    SorobanRpc.isSimulationSuccess(response) &&
+    response.result !== undefined
+  ) {
+    return scValToNative(response.result.retval);
   }
-  return scValToNative(scVal);
+  throw new Error("cannot simulate transaction");
```
Comment (Shaptic) [inline src/helpers/soroban.ts:117]: would be helpful to include the simulation response here (since it may include an error message)

**Review by Shaptic — APPROVED** [review]
LGTM! 

Just to note, though: you aren't capturing the restoration case (`isSimulationRestore`) which would enable a restore-and-retry sequence (but would require another signature, I think, so it probably makes sense to omit here?).

**Comment by aristidesstaffieri** [comment]
> LGTM!
> 
> Just to note, though: you aren't capturing the restoration case (`isSimulationRestore`) which would enable a restore-and-retry sequence (but would require another signature, I think, so it probably makes sense to omit here?).

yeah good call, I think given the scope being more of an example than a solution that handles all scenarios maybe we should leave that out since there isn't a workflow for letting them restore/sign/resubmit at this point. I believe the idea as far as expiration in these examples goes is that we have a WIP example dapp for state expiry and once that's done, we will implement workflows that link out to it in the existing examples in order to handle these cases.

**Pull request merged**